kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -222,7 +223,9 @@ public T getMandatory() {
     if (item != null) {
       return item;
     }
-    throw new KubernetesClientException("Resource not found!", 404, new StatusBuilder().withCode(404).withMessage("Not found.").build());
```

Comment:
Do we not get a nice Status back for a 404? Pretty sure we should do now - maybe this was just a legacy thing?

An example response:

```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "pods \"jimmi\" not found",
  "reason": "NotFound",
  "details": {
    "name": "jimmi",
    "kind": "pods"
  },
  "code": 404
}
```